src/modm/platform/gpio/sam/pin.hpp.in

```diff
@@ -110,6 +110,25 @@ struct OneOfSignals
 	static constexpr bool value = ((std::is_same_v<typename Signal::signal, Signals>) | ...);
 };
 
+// Detect if a pin is inverted
+// Used to support both pins config and gpio classes being passed to GpioSet.
+// Pin config classes don't define isInverted and are treated as non-inverted.
+template<typename T>
+struct is_inverted
+{
+    static constexpr bool value = false;
+};
+
+template<typename T>
+    requires (T::isInverted == true)
+struct is_inverted<T>
+{
+    static constexpr bool value = true;
+};
+
+template<typename T>
+constexpr bool is_inverted_v = is_inverted<T>::value;
```

Comment:
```suggestion
constexpr bool is_inverted_v = T::isInverted;
```
I don't fully understand why this helper is needed?

Reply:
What this does in pseudo-code:
```
if T::isInverted is not a valid expression
    return false
else
    return T::isInverted
```
This is to preserve compatibility with passing these gpio config classes to GpioSet for which `T::isInverted` is not a valid expression.

Reply:
Oh, I see. 👍